terraform/ecs-cluster/s3.tf

```
@@ -0,0 +1,158 @@
+resource "aws_s3_bucket" "container_definitions" {
+  # checkov:skip=CKV_AWS_144: Cross Region Replication not Required
+  # checkov:skip=CKV2_AWS_62: Don't need even notifications
+  # checkov:skip=CKV_AWS_18:  Logging not required
+  bucket = "ecs-container-definitions-${var.application}-${var.environment}"
+  tags   = local.tags
+}
+
+resource "aws_s3_bucket_policy" "container_definitions" {
+  bucket = aws_s3_bucket.container_definitions.id
+  policy = data.aws_iam_policy_document.container_definitions_bucket.json
+}
+
+data "aws_iam_policy_document" "container_definitions_bucket" {
+  statement {
+    effect  = "Deny"
+    actions = ["s3:*"]
+    principals {
+      type        = "*"
+      identifiers = ["*"]
+    }
+
+    resources = [
+      aws_s3_bucket.container_definitions.arn,
+      "${aws_s3_bucket.container_definitions.arn}/*"
+    ]
+
+    condition {
+      test     = "Bool"
+      variable = "aws:SecureTransport"
+      values   = ["false"]
+    }
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:DeleteObject",
+      "s3:DeleteObjectVersion"
+    ]
+
+    principals {
+      type = "AWS"
+      identifiers = [
+        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.application}-${var.environment}-codebase-pipeline-deploy"
+      ]
+    }
+
+    resources = [
+      aws_s3_bucket.container_definitions.arn,
+      "${aws_s3_bucket.container_definitions.arn}/*"
+    ]
+  }
+}
+
+resource "aws_s3_bucket_public_access_block" "container_definitions" {
+  bucket                  = aws_s3_bucket.container_definitions.id
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_versioning" "container_definitions" {
+  bucket = aws_s3_bucket.container_definitions.id
+  versioning_configuration { status = "Enabled" }
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "container_definitions" {
+  bucket = aws_s3_bucket.container_definitions.id
+
+  rule {
+    id     = "NonCurrentVersionsRetention"
+    status = "Enabled"
+
```

```suggestion
filter {}
```
Can we add a `filter {}` block to the S3 lifecycle rules to prevent the Terraform invalid attribute warning